db/compaction_job.cc

```diff
@@ -753,9 +755,7 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
   std::string compression_dict;
   compression_dict.reserve(cfd->ioptions()->compression_opts.max_dict_bytes);
 
-  // TODO(noetzli): check whether we could check !shutting_down_->... only
```

Comment:
With memory_order_relaxed it doesn't seem like a big deal, except if there's false sharing with other stuff in DBImpl.